### ISSUE TITLE
Wrong period for email report

### ISSUE THREAD

**Issue by ma2thieu** [piwik/piwik#11367]
The period in the email report is the same as the one defined on the page `All Websites` and not the one defined by the field `Email schedule` of the report.

**Comment by Findus23** [comment]
If I remember correctly this is intended. If you manually send a report (or download it) it uses the currently selected period, while the daily/weekly/monthly sent report uses the last day/week/month.

**Comment by ma2thieu** [comment]
But the weekly report I received by mail this morning had the right title (Date range: week February 13 – 19, 2017) but the wrong numbers, the numbers were the stats from yesterday (Sunday 19) only, not the stats from last week.

I understand it's intended but I find it confusing that the report I manually send (or download) and the one I receive by mail are not the same.

**Comment by mattab** [comment]
It sounds like there is a bug ie. when you send a report set as "weekly" but for a "day" then the report itself should not mention "weekly" or it would be confusing.

@ma2thieu can you post here a screenshot showing the problem wording?

**Comment by ma2thieu** [comment]
This is the weekly email I received this morning :

![screenshot - 02202017 - 10 24 12 am](https://cloud.githubusercontent.com/assets/1158335/23119168/61e16548-f757-11e6-9c1e-c54536a22d30.png)

And the real weekly stats for these websites : 

![screenshot - 02202017 - 10 25 12 am](https://cloud.githubusercontent.com/assets/1158335/23119167/61e15a4e-f757-11e6-8442-ba665a064d00.png)

**Comment by ma2thieu** [comment]
Actually it's not even the daily stats, the numbers in the report are wrong : 

The email received this monday morning : 

![screenshot - 02272017 - 09 12 27 am](https://cloud.githubusercontent.com/assets/1158335/23353619/9fd0777e-fccd-11e6-9ad0-dc6bdd300b2d.png)


The daily stats from the dashboard : 

![screenshot - 02272017 - 09 12 48 am](https://cloud.githubusercontent.com/assets/1158335/23353624/ab2f0a2c-fccd-11e6-8766-b2c07072e65a.png)


The weekly stats from the dashboard : 

![screenshot - 02272017 - 09 13 13 am](https://cloud.githubusercontent.com/assets/1158335/23353630/b1132cd4-fccd-11e6-8649-1f9d873cf690.png)

**Comment by tsteur** [comment]
For some reasons the pictures don't show up here. What timezone is your website in? Does it work when you create the same report again in a second report?

**Comment by M13iriam** [comment]
I have a similar problem. I have a weekly report and report period set to weekly. My report from this week of july: 8.–14. 2019 shows me data from these periods: 17.-23.12.18, 11.-17..02.19, 08.-14.04.19 and 03.-09.06.19.

**Comment by M13iriam** [comment]
I also configured three test reports, sent to me daily but with different rage of report (day, wee, year) and I still got three times exactly the same report about the data from 22.-28. of July. At least this time for whatever reason the graphs were also showing this range.

**Comment by fvdm** [comment]
I have the same problem. When the email report is generated it takes the period that is set in Matomo when viewing stats (ie. Dashboard). The period you set in the email report settings is ignored or maybe overwritten.

### Steps:

Matomo 3.13.0 on premises

1. Set your overal date range in the Matomo header to Day.
2. Go to Admin > Personal > Email reports.
3. Create a new email report.
4. Set Report Period to Month, select Visitor summary or other useful stats and save it.
5. Click the Download link for the new report.

The line graphs do show the whole month, but the tables only have the stats for the day selected in the overal Matomo date range.

6. Set the date range in the Matomo header to Month.
7. Click the Download link of the report.

Now you get the tables for the whole month.

**Comment by utrautmann** [comment]
> The line graphs do show the whole month, but the tables only have the stats for the day selected in the overal Matomo date range.

Yes, same thing is here. When the date range in the Matomo header is different to the report setting, then the Download or Email-Link-Function generates wrong values.

Another strange behaviour is, when you set following
![grafik](https://user-images.githubusercontent.com/25771066/96279949-a4ab3100-0fd7-11eb-994e-da4edc15e71c.png)

Then the line graphs shows the last 6 days
![grafik](https://user-images.githubusercontent.com/25771066/96280594-6cf0b900-0fd8-11eb-8d05-462500ec449e.png)

**Comment by Findus23** [comment]
Hi,

As far as I know the "Download or Email-Link-Function" always take the currently selected date range in Matomo, not what is configured for the Report.

**Comment by utrautmann** [comment]
Hi, 
yes, it has been like this for years. But why? It's irritating from a usability point of view.
Anyone I know who uses this feature for the first time or after long time believes there is a bug in their report configuration.
It takes time to figure out that it might have something to do with the date range above.

Is it not possible to set the period settings from the report in the two links and not those from the calendar?

**Comment by mattab** [comment]
I reckon more users would maybe want to see the report use the period as it was setup (otherwise it would be confusing also). But for sure it's not very usable and understand your point. Maybe it would work to give users the option or show 2 download and 2 email links instead of 1 each...

**Comment by tsteur** [comment]
same here: https://github.com/matomo-org/matomo/issues/16644

**Comment by CoreyBurger** [comment]
Here from #16644. I suggest that the download or send report link use the report as setup, as from a new users perspective, it looks like a bug and my use case was "Is this report working correctly? Does it show what I want?". 

To solve the use case of "give me a scratch report of the selection on the main screen", I would add a button there to do that on the main screen.

**Comment by tsteur** [comment]
In https://github.com/matomo-org/matomo/issues/17791 was suggested to show a tooltip or a footer message explaining this.
